htlcswitch/link.go

```diff
@@ -102,6 +102,11 @@ type ChannelLinkConfig struct {
 	// with the debug htlc R-Hash are immediately settled in the next
 	// available state transition.
 	DebugHTLC bool
+
+	// SyncStates is used to indicate that we need send the channel
+	// reestablishment message to the remote peer. It should be done if our
+	// clients have been restarted, or remote peer have been reconnected.
+	SyncStates bool
```

Comment:
I don't think we need this parameter at all. Upon the initialization of a channel, retransmission should be performed _unconditionally_. If both sides are in sync, then the process will simply be a noop.